Show whitespace in hunk while "show_whitespaces": "selection"

rien7 opened the pull request:
Closes #25250 

Release Notes:

- Show whitespace in hunk while setting `"show_whitespaces": "selection"`
- add `show_whitespaces_in_git_diff` settings. this setting will override the `show_whitespaces` setting in git diff hunk

![GkPQLFea0AANrYj](https://github.com/user-attachments/assets/2132b5c1-8812-4c2b-b92e-3a1d049d58de)

rien7 committed:
feat: show whitespace in git diff hunk

0x2CA commented:
Very nice, I think a configuration should be needed as different people have different preferences, some want it to show when diff, some don't.

rien7 committed:
feat: add `show_whitespaces_in_git_diff` settings

if `show_whitespaces_in_git_diff` is true, this setting will show whitespaces in git diff hunks, overriding the global `show_whitespaces` setting.

maxdeviant commented:
Thank you for the PR, but we don't want to add this functionality at this time, as it muddies the narrative around the settings.

rien7 commented:
@maxdeviant Thanks for your feedback! I understand the concern about muddying the narrative around the settings. However, I believe that displaying whitespace characters in git diff is a necessary feature for better readability and accuracy when reviewing changes. Would you be open to discussing alternative ways to integrate this functionality, perhaps as a default behavior without a setting?
